runtimes/protocol/lsp.ts

```diff
@@ -40,6 +41,47 @@ export class AutoParameterStructuresProtocolRequestType<P, R, PR, E, RO>
     }
 }
 
+/**
+ * Extra Client information passed during initialization handshake from Client to Server.
+ */
+export interface AWSClientInitializationOptions {
+    /**
+     * Custom client branding information.
+     */
+    product?: {
+        /**
+         * Product name, may contain spaces.
+         */
+        name: string
+        version: string
+    }
+    /**
+     * Client host platform information.
+     */
+    platform?: {
```

Comment:
What is the purpose of platform? Is the "client host platform" ever different from the platform the runtime runs on?

Reply:
`platform` here refers to the IDE or Editor, in which Client runs. I'll clarify the naming, platform maybe not the best.

From language server process pov, platform is host machine, but not IDE.